added tests

metelska-anastasiia opened the pull request:
for ShoppingCart, CartItem, User, Order, OrderItem entities (repository, service, controller)
<img width="604" alt="Снимок экрана 2023-11-08 в 11 43 33" src="https://github.com/metelska-anastasiia/online-book-store/assets/117344368/93409edc-db6d-47cb-a945-0d204633d577">

metelska-anastasiia committed:
added tests for ShoppingCart, CartItem, User, Order, OrderItem entities (repository, service, controller)

Ivan95kos commented on src/test/java/mate/academy/controller/AuthenticationControllerTest.java:
```diff
@@ -0,0 +1,166 @@
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class AuthenticationControllerTest {
```
```suggestion
class AuthenticationControllerIntegrationTest {
```

Ivan95kos commented on src/test/java/mate/academy/controller/AuthenticationControllerTest.java:
```diff
@@ -0,0 +1,166 @@
+
+
+
+
```
lines
```suggestion
```

Ivan95kos commented on src/test/java/mate/academy/controller/OrderControllerTest.java:
```diff
@@ -0,0 +1,211 @@
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class OrderControllerTest {
```
```suggestion
class OrderControllerIntegrationTest {
```

Ivan95kos commented on src/test/java/mate/academy/controller/ShoppingCartControllerTest.java:
```diff
@@ -0,0 +1,206 @@
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class ShoppingCartControllerTest {
```
the same

Ivan95kos commented on src/test/java/mate/academy/repository/BookRepositoryTest.java:
```diff
@@ -2,30 +2,43 @@
 
 @DataJpaTest
 @AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
 class BookRepositoryTest {
```
also
```suggestion
class BookRepositoryIntegrationTest {
```

Ivan95kos commented on src/test/java/mate/academy/repository/CartItemRepositoryTest.java:
```diff
@@ -0,0 +1,82 @@
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class CartItemRepositoryTest {
```
also

Ivan95kos commented on src/test/java/mate/academy/repository/OrderRepositoryTest.java:
```diff
@@ -0,0 +1,108 @@
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class OrderRepositoryTest {
```
+

Ivan95kos commented on src/test/java/mate/academy/repository/ShoppingCartRepositoryTest.java:
```diff
@@ -0,0 +1,74 @@
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class ShoppingCartRepositoryTest {
```
+

Ivan95kos commented on src/test/java/mate/academy/repository/UserRepositoryTest.java:
```diff
@@ -0,0 +1,60 @@
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class UserRepositoryTest {
```
also

Ivan95kos commented on src/test/java/mate/academy/repository/BookRepositoryTest.java:
```diff
@@ -65,4 +78,82 @@ void findAll_WithPagination_ShouldReturnPage() {
+
+        List<Book> expected = new ArrayList<>();
+        expected.add(
+                new Book().setId(1L)
+                        .setTitle("Book 1")
+                        .setAuthor("Author 1")
+                        .setIsbn("ISBN-123456")
+                        .setPrice(BigDecimal.valueOf(100))
+                        .setDescription("Description for Book 1")
+                        .setCoverImage("image1.jpg")
+                        .setCategories(Set.of(new Category()
+                                .setId(1L)
+                                .setName("Poetry")
+                                .setDescription("Poems that you will love"))
+                        )
```
it is better to move the common data to the private methods,
fix it everywhere

Ivan95kos commented on src/test/java/mate/academy/service/OrderServiceImplTest.java:
```diff
@@ -0,0 +1,258 @@
+
+    @BeforeEach
+    void setUp() {
+        user = new User()
+                .setId(USER_ID)
+                .setFirstName("John")
+                .setLastName("Doe")
+                .setPassword("test")
+                .setEmail("john@doe.com");
+        book = new Book()
+                .setId(1L)
+                .setIsbn("123456789")
+                .setTitle("Title 1")
+                .setAuthor("Author 1")
+                .setPrice(BigDecimal.valueOf(100))
+                .setDescription("Description 1")
+                .setCoverImage("image1");
+        orderItem = new OrderItem()
+                .setId(1L)
+                .setQuantity(1)
+                .setBook(book)
+                .setPrice(BigDecimal.valueOf(100));
+        order = new Order()
+                .setUser(user)
+                .setOrderDate(LocalDateTime.of(2023, 11, 4, 11,30,1))
+                .setTotal(BigDecimal.valueOf(100))
+                .setStatus(Order.Status.NEW)
+                .setOrderItems(Set.of(orderItem))
+                .setId(1L);
+        cartItem = new CartItem()
+                .setBook(book)
+                .setQuantity(1)
+                .setId(1L);
+        orderItemResponseDto = new OrderItemResponseDto()
+                .setId(1L)
+                .setQuantity(1)
+                .setBookId(1L);
```
Such code is better to read, 
and you can use these methods wherever you need
```suggestion
        user = prepareUser();
        book = prepareBook();
        orderItem = prepareOrderItem();
        order = prepareOrder();
        cartItem = prepareCartItem();
        orderItemResponseDto = prepareOrderItemResponseDto();
```

Ivan95kos commented on src/test/java/mate/academy/service/OrderServiceImplTest.java:
```diff
@@ -0,0 +1,258 @@
+    private OrderServiceImpl orderService;
+    @Mock
+    private Authentication authentication;
+    private User user;
+    private Book book;
+    private OrderItem orderItem;
+    private Order order;
+    private CartItem cartItem;
+    private OrderItemResponseDto orderItemResponseDto;
```
It is better not to use the general test data
```suggestion
```

Ivan95kos commented on src/test/java/mate/academy/service/OrderServiceImplTest.java:
```diff
@@ -0,0 +1,258 @@
+    private OrderMapper orderMapper;
+    @Mock
+    private OrderItemMapper orderItemMapper;
+    @InjectMocks
+    private OrderServiceImpl orderService;
+    @Mock
+    private Authentication authentication;
```
change the order
```suggestion
    @Mock
    private Authentication authentication;
    @InjectMocks
    private OrderServiceImpl orderService;
```

Ivan95kos reviewed (changes requested):
left a few comments

metelska-anastasiia committed:
fixed all comments

Sanderfuck commented on src/test/java/mate/academy/controller/AuthenticationControllerIntegrationTest.java:
```diff
@@ -0,0 +1,158 @@
+    @Autowired
+    private JwtUtil jwtUtil;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext
+    ) {
```
Why you Autowire here?

Sanderfuck commented on src/test/java/mate/academy/controller/AuthenticationControllerIntegrationTest.java:
```diff
@@ -0,0 +1,158 @@
+        teardown(dataSource);
+    }
+
+    @BeforeEach
+    void setUp(@Autowired DataSource dataSource) {
+        setupDatabase(dataSource);
+    }
+
+    @AfterEach
+    void afterEach(@Autowired DataSource dataSource) {
+        teardown(dataSource);
+    }
```
I think exist better approach for init and clean database

Sanderfuck commented on src/test/java/mate/academy/controller/AuthenticationControllerIntegrationTest.java:
```diff
@@ -0,0 +1,158 @@
+    @Test
+    @DisplayName("Register new user")
+    void register_validRequest_Success() throws Exception {
+        UserRegistrationRequest request = new UserRegistrationRequest()
+                .setEmail("newuser@test.com")
+                .setPassword("Newuser12345")
+                .setRepeatPassword("Newuser12345")
+                .setFirstName("User")
+                .setLastName("New");
+
+        UserResponseDto expected = new UserResponseDto()
+                .setId(3L)
+                .setEmail(request.getEmail())
+                .setFirstName(request.getFirstName())
+                .setLastName(request.getLastName());
+
```
You can create template request, users, etc in some common file and use it every test what you need for increase readability

Sanderfuck commented on src/test/java/mate/academy/controller/BookControllerIntegrationTest.java:
```diff
@@ -308,4 +284,16 @@ void updateBook_WithValidIdAndCreateBookRequestDto_Success() throws Exception {
         EqualsBuilder.reflectionEquals(expected, actual);
     }
+
+    private BookDto prepareBookDto() {
+        return new BookDto()
+                .setId(1L)
+                .setTitle("Book 1")
+                .setAuthor("Author 1")
+                .setIsbn("ISBN-123456")
+                .setPrice(BigDecimal.valueOf(100))
+                .setDescription("Description for Book 1")
+                .setCoverImage("image1.jpg")
+                .setCategoryIds(Set.of());
+    }
```
Replace to common class with templates

metelska-anastasiia committed:
Added new class with templates
Replaced Autowire from method variables to class variables in all controller tests
